tests/test_convert_lit_checkpoint.py

```diff
@@ -0,0 +1,78 @@
+from pathlib import Path
+from urllib.request import urlretrieve
+
+import pytest
+import torch
+
+wd = Path(__file__).parent.parent.absolute()
+
+
+@torch.inference_mode()
+def test_against_original_falcon_40b():
+    file_path = wd / "tests" / "original_falcon_40b.py"
+    url = "https://gist.githubusercontent.com/carmocca/feed39b1bc65a29f73c1cecc58a01167/raw/a9a65f2b93716b3c09ec9f354d535ae5953de08f/original_falcon_40b.py"
+    if not file_path.is_file():
+        urlretrieve(url=url, filename=file_path)
+
+    from tests.original_falcon_40b import RWConfig, RWForCausalLM
+    from lit_gpt import Config, GPT
+    from scripts.convert_lit_checkpoint import copy_weights_falcon
+    from scripts.convert_hf_checkpoint import copy_weights_falcon as copy_weights_falcon_hf
+
+    # the unchanged model config
+    our_config = Config.from_name("falcon-40b", n_layer=2, n_head=8, n_query_groups=4, n_embd=32)
+    theirs_config = RWConfig(
+        hidden_size=32, n_head=8, n_head_kv=4, n_layer=2, parallel_attn=True, vocab_size=65024, bias=False
+    )
+
+    # need to begin with weight initialized from a ground truth source
+    ground_truth_their_model = RWForCausalLM(theirs_config)
+    ground_truth_their_state_dict = ground_truth_their_model.state_dict()
```

Comment:
Why is this necessary? Conversion should work if you start with our model.

Reply:
improved in https://github.com/Lightning-AI/lit-gpt/pull/273/commits/ee894af7f93c27691fbcbfbe5da3b1349355fe5a